mlflow/pyfunc/model.py

```diff
@@ -327,12 +333,40 @@ def _load_pyfunc(model_path):
         python_model = cloudpickle.load(f)
 
     artifacts = {}
+    hf_prefix = "hf:/"
     for saved_artifact_name, saved_artifact_info in pyfunc_config.get(
         CONFIG_KEY_ARTIFACTS, {}
     ).items():
-        artifacts[saved_artifact_name] = os.path.join(
-            model_path, saved_artifact_info[CONFIG_KEY_ARTIFACT_RELATIVE_PATH]
-        )
+        if CONFIG_KEY_ARTIFACT_RELATIVE_PATH in saved_artifact_info:
+            artifacts[saved_artifact_name] = os.path.join(
+                model_path, saved_artifact_info[CONFIG_KEY_ARTIFACT_RELATIVE_PATH]
+            )
+        else:
+            artifact_uri = saved_artifact_info[CONFIG_KEY_ARTIFACT_URI]
+            if artifact_uri.startswith(hf_prefix):
+                try:
+                    from huggingface_hub import snapshot_download
+                except ImportError as e:
+                    raise MlflowException(
+                        "Failed to import huggingface_hub. Please install huggingface_hub to "
+                        f"download snapshot with artifact_uri: {artifact_uri}. Error: {e}"
+                    )
+                try:
+                    hf_uri = artifact_uri[len(hf_prefix) :]
+                    splits = hf_uri.split("?")
+                    repo_id = splits[0]
+                    kwargs = (
+                        dict([param.split("=") for param in splits[1].split("&")])
+                        if len(splits) > 1
+                        else {}
+                    )
+                    snapshot_location = snapshot_download(repo_id=repo_id, **kwargs)
```

Comment:
This will download the files and create a simlink to the the local cache only on load, right?

This requires a few assumptions:
1. The serving location has access to the open internet to connect to huggingface_hub directly
2. The repo is consistent over time (a new revision hasn't been pushed)
3. The artifacts are consistently available on huggingface_hub (the repo hasn't been deleted)
4. The hub is up and capable of responding to requests for download
 
I think it would be safer to actually save the artifacts directly when the function ``_save_model_with_class_artifacts_params()`` is called. That way we can avoid all of those issues and maintain similar behavior to all other MLflow implementations. 

Alternative design:
- User defines a pre-trained model to download from a huggingface_hub repo via ``"hf:/"`` 
- Check if artifact is present in cache. (this should be automatic; will need to verify this functionality)
  - If it is, use that instance to directly copy the artifact to the artifact store
  - If it isn't, disable the cache during the snapshot download 
 ``snapshot_download(repo_id=repo_id, local_dir=<tmp_artifacts_dir_path>, local_dir_use_simlinks=False, **kwargs)``

On load, the model will use the pre-existing implementations for handling artifacts.